Doc/library/sqlite3.rst

```diff
@@ -47,84 +47,148 @@ This document includes four main sections:
-:file:`example.db` file::
+In this tutorial, you will learn the basics of the :mod:`!sqlite3` API
+by creating an on-disk movie database :file:`tutorial.db`
+an filling it with meta-data about Monty Python movies.
+A fundamental understanding of database concepts,
+like `transactions`_ and `cursors`_, is assumed.
+
+To start working with an SQLite database,
+you must first open a connection to it.
+Call the :func:`sqlite3.connect` function
+to create a connection to the database :file:`tutorial.db`::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("tutorial.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+SQLite will implicitly create an empty database file :file:`tutorial.db`
+in the current working directory if it does not exist.
+The returned :class:`Connection` object ``con``
+represents the connection to the on-disk database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+use a database cursor.
+Call :meth:`~Connection.cursor` on ``con`` to create
+:class:`Cursor` ``cur``::
 
    cur = con.cursor()
 
-   # Create table
-   cur.execute('''CREATE TABLE stocks
-                  (date text, trans text, symbol text, qty real, price real)''')
+Now, create a database table ``movie`` with columns for title,
+year released, and `IMDB`_ review score.
+The `flexible typing`_ feature of SQLite makes typenames optional;
+for simplicity, just use column names in the table declaration.
+Execute the SQL statement
+by calling :meth:`~Cursor.execute` on ``cur``::
 
-   # Insert a row of data
-   cur.execute("INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14)")
+   cur.execute("CREATE TABLE Movie(Title, Year, Score)")
 
-   # Save (commit) the changes
-   con.commit()
+Verify that the table has been created by running a query against it.
+Execute the query by calling :meth:`~Cursor.execute` on ``cur``,
+store the result in a variable ``res``,
+and call :meth:`~Cursor.fetchone` on ``res`` to fetch the first
+(and only) row from the query::
 
-   # We can also close the connection if we are done with it.
-   # Just be sure any changes have been committed or they will be lost.
-   con.close()
+   >>> res = cur.execute("SELECT count(rowid) FROM Movie")
+   >>> res.fetchone()
+   (0,)
 
-The saved data is persistent: it can be reloaded in a subsequent session even
-after restarting the Python interpreter::
+The result is a one-item :class:`tuple`:
+one row, with one column.
+It correctly shows a row count of zero.
 
-   import sqlite3
-   con = sqlite3.connect('example.db')
-   cur = con.cursor()
+Now, add two rows of data supplied as SQL literals
+by executing an ``INSERT`` statement,
+once again by calling :meth:`~Cursor.execute` on ``cur``::
```

Comment:
```suggestion
by executing an ``INSERT`` statement with :meth:`cur.execute() <Cursor.execute>`::
```

Reply:
I think leaving the "once again" is nice, because it reminds the user that we should now be familiar with the `execute` method. Maybe it is not needed.

Reply:
Seems fair enough to me